app/store/services/event.js

```diff
@@ -0,0 +1,75 @@
+import { coordinates, markers } from "../sagas/data/data";
+import positionHelper from "./helpers/gps";
+import { convertGradeToColor, convertPercentToGrade } from "./helpers/event";
+
+const compose = (...fns) => x => fns.reduceRight((v, f) => f(v), x);
+
+const addData = (...edges) => {
+  let distanceDone = 0;
+  return edges.reduce((accu, edge, index) => {
+    const length = positionHelper.computeDistance(edge);
+    distanceDone += length;
+    const elevation = (edge.dest.altitude - edge.src.altitude) / 1000;
+    const percent = elevation / length * 100;
+    const grade = convertPercentToGrade(percent);
+    const color = convertGradeToColor(grade);
+
+    const enhancedEdge = {
+      edge,
+      length,
+      percent,
+      index,
+      distanceDone,
+      grade,
+      color
+    };
+    return [...accu, enhancedEdge];
+  }, []);
+};
+
+const groupBy = (...edges) => {
```

Comment:
duplicate?

Reply:
Cleaning in progress! At the end, only one will....